initial-checks.sh

```diff
@@ -11,9 +11,21 @@ uname -s | grep -i linux
 check_for_failure "Please run this script on Linux"
 
 if [ "$EUID" -ne 0 ];then
-  check_for_failure "Please run as root or with sudo"
+  echo "Please run as root or with sudo"
+  exit 1
```

Comment:
Oh interesting. I was under the assumption that test would update the return code immediately. Ok cool. Please disregard my previous comment since your code patches this bug. I can swing back around to this later.